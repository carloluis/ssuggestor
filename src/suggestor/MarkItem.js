import React from 'react';

const MarkItem = ({ item, search }) => {
	if (!search) {
		return <a>{ item }</a>;
	}

	let index = item.toLowerCase().indexOf(search.toLowerCase());
	if (index === -1) {
		return <a>{ item }</a>;	
	}

	let searchLength = search.length;

	return (
		<a>
			<span>
				{ item.substr(0, index) }
				<strong>{ item.substr(index, searchLength) }</strong>
				{ item.substr(index+searchLength, item.length) }
			</span>
		</a>
	);
};
MarkItem.propTypes = {
	item: React.PropTypes.string.isRequired,
	search: React.PropTypes.string.isRequired
};

export default MarkItem;