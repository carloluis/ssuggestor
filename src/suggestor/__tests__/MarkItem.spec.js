import React from 'react';
import MarkItem from '../MarkItem';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

const PROPS = {
	item: {
		word: 'https://randomword.com/',
		index: 0
	},
	search: ''
};

describe('<MarkItem />', () => {
	it('snapshot - plain item <a/> when no search pattern', () => {
		const tree = renderer.create(<MarkItem {...PROPS} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('snapshot - plain item <a/> when no match', () => {
		const props = { item: { ...PROPS.item, index: -1 }, search: 'x' };

		const tree = renderer.create(<MarkItem {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('snapshot - render pattern inside <strong/> when pattern matched', () => {
		const props = { item: { word: 'temporise', index: 5 }, search: 'rise' };

		const tree = renderer.create(<MarkItem {...props} />);

		expect(tree).toMatchSnapshot();
	});
});
