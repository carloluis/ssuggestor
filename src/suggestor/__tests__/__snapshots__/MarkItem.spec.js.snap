// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkItem /> snapshot - plain item <a/> when no match 1`] = `
<a>
  https://randomword.com/
</a>
`;

exports[`<MarkItem /> snapshot - plain item <a/> when no search pattern 1`] = `
<a>
  https://randomword.com/
</a>
`;

exports[`<MarkItem /> snapshot - render pattern inside <strong/> when pattern matched 1`] = `
<a>
  <span>
    tempo
    <strong>
      rise
    </strong>
    
  </span>
</a>
`;
